modfiles/data/calculation/matrix_solver.lua

```diff
@@ -203,13 +203,31 @@ function matrix_solver.get_matrix_solver_metadata(subfactory_data)
     return result
 end
 
+function matrix_solver.transpose(m)
+    local transposed = {}
+
+    if #m == 0 then
+        return transposed
+    end
+
+    for i=1, #m[1] do
+        local row = {}
+        for j=1, #m do
+            table.insert(row, m[j][i])
+        end
+        table.insert(transposed, row)
+    end
+    return transposed
+end
+
 function matrix_solver.get_linear_dependence_data(subfactory_data, matrix_metadata)
     local num_rows = matrix_metadata.num_rows
     local num_cols = matrix_metadata.num_cols
 
     local linearly_dependent_recipes = {}
     local linearly_dependent_items = {}
     local allowed_free_items = {}
```

Comment:
For the purposes of this PR it's fine to delete the original `allowed_free_items` logic, save the fast result in `allowed_free_items`, and return the fast result.